modules/distributor/distributor.go

```diff
@@ -351,6 +362,12 @@ func (d *Distributor) PushBatches(ctx context.Context, batches []*v1.ResourceSpa
 		d.generatorForwarder.SendTraces(ctx, userID, keys, rebatchedTraces)
 	}
 
+	if len(d.cfg.Forwarders) > 0 && len(d.overrides.Forwarders(userID)) > 0 {
```

Comment:
nit: seems unnecessary to check len of `d.cfg.Forwarders`

Reply:
len(d.cfg.Forwarder) can be 0 in a situation where there is a forwarder defined for a particular tenant in the `overrides.yaml` config, but forwarders are defined in the distributor's `config.yaml`.

Also, this check serves as an additional safety net, it ensures that the feature is completely toggled off when there are no forwarders defined in distributor's `config.yaml`. If there were a bug or anything, it guarantees a super simple way to shut down this feature.